docs/ROADMAP.md

```diff
@@ -7,27 +7,33 @@ category: Roadmap
 
 This document defines the roadmap for TiKV development.
 
-- [ ] Raft
-    - [x] Region merge
-    - [ ] Local read thread
-    - [ ] Multi-thread raftstore
-    - [x] None voter
-    - [x] Pre-vote
-    - [ ] Multi-thread apply pool
-    - [ ] Split region in batch
-    - [ ] Raft Engine
-- [x] RocksDB 
-    - [x] DeleteRange
-    - [ ] BlobDB 
-- [x] Transaction
-    - [x] Optimize transaction conflicts
-    - [ ] Distributed GC
-- [x] Coprocessor
-    - [x] Streaming
-- [ ] Tool
-    - [x] Import distributed data
-    - [ ] Export distributed data
-    - [ ] Disaster Recovery
-- [ ] Flow control and degradation
-- [ ] Client
-    - [ ] TiKV client (Rust crate)
+## Raft
+- [x] Region Merge - Merge small Regions together to reduce overhead
+- [x] Local Read Thread - Process read requests in a local read thread
+- [x] Split Region in Batch - Speed up Region split for large Regions
+- [x] Raft Learner - Support Raft learner to smooth the configuration change process
+- [x] Raft Pre-vote - Support Raft pre-vote to avoid unnecessary leader election on network isolation
+- [ ] Joint Consensus - Change multi members safely.
+- [ ] Multi-thread Raftstore - Process Region Raft logic in multiple threads
+- [ ] Multi-thread apply pool - Apply Region Raft committed entries in multiple threads
+
+## Engine
+- [ ] Titan - Separate large key-values from LSM-Tree
+- [ ] Pluggable Engine Interface - Clean up the engine wrapper code and provide more extendibility
+
+## Transaction
+- [x] Optimize transaction conflicts
+- [ ] Distributed GC - Distribute MVCC garbage collection control to TiKV
+
+## Coprocessor
+- [ ] Coprocessor Chunk Execution - Process data in chunk to improve performance
```

Comment:
Streaming is used for returning large data set (e.g. `select * from table`) from TiKV. Enable it can cut large data set into several small chunks in order to be more friendly with memory usage.